src/logic/hooks/useEstimateSafeCreationGas.tsx

```diff
@@ -20,40 +20,48 @@ type SafeCreationEstimationResult = {
 }
 
 const { nativeCoin } = getNetworkInfo()
-
+const DEFAULT_GAS = '< 0.001'
 export const useEstimateSafeCreationGas = ({
   addresses,
   numOwners,
   safeCreationSalt,
 }: EstimateSafeCreationGasProps): SafeCreationEstimationResult => {
-  const [gasEstimation, setGasEstimation] = useState<SafeCreationEstimationResult>({
+  const [gasEstimation, setGasEstimation] = useState({
     gasEstimation: 0,
-    gasCostFormatted: '< 0.001',
     gasLimit: 0,
   })
+  const [gasCostFormatted, setGasCostFormated] = useState<string>(DEFAULT_GAS)
   const userAccount = useSelector(userAccountSelector)
 
+  useEffect(() => {
+    const estimateGasPrice = async () => {
+      const gasPrice = await calculateGasPrice()
+      const estimatedGasCosts = gasEstimation.gasEstimation * parseInt(gasPrice, 10)
+      const gasCost = fromTokenUnit(estimatedGasCosts, nativeCoin.decimals)
+      const gasAmount = formatAmount(gasCost)
+      setGasCostFormated(gasAmount)
+    }
+    if (gasEstimation.gasEstimation != 0) {
```

Comment:
Why does the hook have to be changed at all? It is triggered because we pass an object that is recreated every render

Reply:
I think the solution isn't this `if` but the dependency list.

Jaumpi moved the gas price request into a separate useEffect with just one dependency: `gasEstimation`

Before it was one big useEffect depending on `numOwners, userAccount, safeCreationSalt, addresses` (each of which would trigger an oracle request).

Reply:
`gasEstimation.gasEstimation` isn't  an object, it's a number. As soon as we know how much gas a tx will take, we call the oracle.

Does it answer your concern @mikheevm? Could you unblock the PR please?

Reply:
> gasEstimation.gasEstimation isn't an object, it's a number. As soon as we know how much gas a tx will take, we call the oracle.

I was talking about the object that is passed an argument, not about this one

> Does it answer your concern @mikheevm? Could you unblock the PR please?

No, it doesn't, if one of the hook parameters change it won't estimate gas anymore

Reply:
Ahh I now understand what you're talking about, so yeah, this kind of works but the gas price is volatile lately and I'm not sure if we only want to calculate it once

Reply:
But still, I don't get why it has to be moved. It's not a thing that you can request once and forget about it

Reply:
And another thing: the gas price is volatile, so probably we'd want to fetch it in an interval

Reply:
I think see your point now: the gas can stay the same but the price could change in the meanwhile?

Reply:
Sorry, I might have overcomplicated things. Let's forget about everything for a moment.

My main concern about the PR is that we have a useEffect hook that is run on a page with no possible changes in data. The solution to this problem is to figure out why this happens, not to add a hacky if statement and prevent it from running (with this solution gas estimation is still run btw)

Reply:
I still don't get it then.
The "hacky" `if` is there just to avoid requesting gas initially, when we don't have a gas amount yet (0 is the default value). As soon as we know how much gas a tx costs, we request the price.

Reply:
I think we should stop flaming in this PR. Maybe it's better if you guys, Mikhail and Juampi, have a quick call to find the best solution.

Reply:
https://github.com/gnosis/safe-react/compare/bug/2350-unnecessary-gas-price-estimations-during-creation?expand=1

Check changed files, hope this gives a better understanding of the real problem

Happy to jump on a call to explain more

Reply:
I've looked further into it and the problem is harder than I initially thought:
1. The review component mounts/unmounts for some reason
2. The gas estimation logic is buggy. We estimate gas and set it as form value, but the gas estimation depends on form values.

I can take it if you want.

Reply:
@mikheevm that's why I've been pushing for just a throttle in the fetchGas function, that would be a versatile fix.
Even if we fix the re-rendering issue, it might be too brittle in the long run.
Anyway, since you're already looking into it and have a better understanding, sure, go for it. Juampi is on holiday today.